src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -243,13 +283,17 @@ public QueryExecutor(DbExceptionParser dbExceptionParser,
             string dataSourceName,
             List<string>? args = null)
         {
+            Stopwatch queryExecutionTimer = new();
+            queryExecutionTimer.Start();
             await conn.OpenAsync();
             DbCommand cmd = PrepareDbCommand(conn, sqltext, parameters, httpContext, dataSourceName);
 
             try
             {
                 using DbDataReader dbDataReader = ConfigProvider.GetConfig().MaxResponseSizeLogicEnabled() ?
                     await cmd.ExecuteReaderAsync(CommandBehavior.SequentialAccess) : await cmd.ExecuteReaderAsync(CommandBehavior.CloseConnection);
+                queryExecutionTimer.Stop();
+                AddDbExecutionTimeToMiddlewareContext(queryExecutionTimer.ElapsedMilliseconds);
```

Comment:
Your code block above had `Timer.Stop()` in the catch block. Is this `.Stop()` missed if an exception is thrown when executing query?

Reply:
fixed.